lib/Frontend/Frontend.cpp

```diff
@@ -312,16 +310,14 @@ static bool shouldTreatSingleInputAsMain(InputFileKind inputKind) {
 }
 
 bool CompilerInstance::setUpInputs() {
-  // Adds to InputSourceCodeBufferIDs, so may need to happen before the
-  // per-input setup.
-  const Optional<unsigned> codeCompletionBufferID = setUpCodeCompletionBuffer();
-
   for (const InputFile &input :
        Invocation.getFrontendOptions().InputsAndOutputs.getAllInputs())
     if (setUpForInput(input))
       return true;
 
   // Set the primary file to the code-completion point if one exists.
+  // Note that this must happen after the regular inputs have been set up.
```

Comment:
Extreme nitpick: An English teacher once drilled into me that "Note that" can always be dropped from any sentence. And another English major once told me to always follow "this" with a noun. Thanks for your patience!

Reply:
It would help a newbie to the code base to expand the comment to explain *why* this setting of the primary file must happen after the regular inputs are set up, rather than before or at the same time, whatever that means.